### ISSUE TITLE
Nullable TimeDimension creates incorrect GQL errors

### ISSUE THREAD

**Issue by iiq374** [cube-js/cube#6972]
**Describe the bug**
Where a date or datetime is in GQL whenever that is an optional field an error is returned EG:
Message: Cannot return null for non-nullable field TimeDimension.day
It seems to be because although the TimeDimension is correctly created as optional in the GQL Schema, the properties are not - so querying for the property creates the error.

**To Reproduce**
Create a cube with a time dimension:
```    
managementStartDate: {
      sql: `management_start_date`,
      type: `time`,
    },
```
Query that field via GQL interface:
managementStartDate {
  day
}

Receive errors per null field.

**Expected behavior**
Where the underlying data is null there should be no error included in the errors array.

**Screenshots**
![image](https://github.com/cube-js/cube/assets/5369087/489d307f-82e5-4cf3-ab3e-0075c6304840)

**Minimally reproducible Cube Schema**
In case your bug report is data modelling related please put your minimally reproducible Cube Schema here.
You can use selects without tables in order to achieve that as follows.

```javascript
cube(`Orders`, {
  sql: `
  select 1 as id, null as shippedDate
  UNION ALL
  select 2 as id, '2023-07-01 12:00:00.000' as shippedDate
  `,
  measures: {
    count: {
      type: `count`,
    },
  },
  dimensions: {
    shippedDate: {
      sql: `shippedDate`,
      type: `time`,
    },
  },
});
```

**Version:**
0.33.39

**Additional context**

**Comment by igorlukanin** [comment]
Still reproduces on Cube v0.36.7:

<img width="1392" alt="Screenshot 2024-10-09 at 00 55 44" src="https://github.com/user-attachments/assets/08dd1c3f-5127-469a-ad01-ed03e0403805">

**Comment by kevinleeTCA** [comment]
Yes, confirmed this is still happening on latest version Cube v0.36.7:

**Comment by rkshvish** [comment]
Hi @paveltiunov - I did small experiment by changing in the file `packages/cubejs-api-gateway/src/graphql.ts`  timeDimension Object like below and removed `notNull` condition and it worked.
```
export const TimeDimension = objectType({
  name: 'TimeDimension',
  definition(t) {
    t.field('value', {
      type: 'DateTime',
    });
    t.nonNull.field('second', {
      type: 'DateTime',
    });
    t.nonNull.field('minute', {
      type: 'DateTime',
    });
    t.nonNull.field('hour', {
      type: 'DateTime',
    });
    t.nonNull.field('day', {
      type: 'DateTime',
    });
    t.nonNull.field('week', {
      type: 'DateTime',
    });
    t.nonNull.field('month', {
      type: 'DateTime',
    });
    t.nonNull.field('quarter', {
      type: 'DateTime'
    });
    t.nonNull.field('year', {
      type: 'DateTime',
    });
  },
});
```

Will this have any impact on any other thing or not, just wanted to know that. If it does not affect anything can this be changed for all time dimesion granularity types?

**Comment by shan63** [comment]
This is causing problems for us too. Is there a reason for requiring time dimensions to always be non-null?

**Comment by scastiel** [comment]
Hi! We can't use GraphQL to query views that contain nullable time dimensions, which is a pretty huge blocker. Any progress on this issue? Thanks!

**Comment by alena-kono** [comment]
Hey there! We’ve also been running into a bit of a roadblock with GraphQL as it doesn’t support querying views that have nullable time dimensions. This has been a pretty big blocker for us, so I just wanted to check if there’s been any progress or updates on that front. Thanks a lot!

**Comment by DarrellKilian** [comment]
Also having to find workarounds for this that should not be necessary.
